File: Monica for wear/src/main/java/takagi/ru/monica/wear/data/SecureItem.kt
package takagi.ru.monica.wear.data

import androidx.room.Entity
import androidx.room.PrimaryKey
import java.util.Date

/**
 * 统一的安全数据项实体 (Wear版本 - 仅支持TOTP)
 * 复用主应用的数据模型，但仅处理TOTP类型
 */
@Entity(tableName = "secure_items")
data class SecureItem(
    @PrimaryKey(autoGenerate = true)
    val id: Long = 0,
    
    // 通用字段
    val itemType: ItemType,  // 数据类型 (Wear版本仅使用TOTP)
    val title: String,       // 标题/名称
    val notes: String = "",  // 备注
    val isFavorite: Boolean = false,
    val sortOrder: Int = 0,  // 排序顺序
    val createdAt: Date = Date(),
    val updatedAt: Date = Date(),
    
    // 类型特定数据(JSON格式存储)
    val itemData: String,    // 存储TOTP数据(JSON)
    
    // 图片附件路径(Wear版本不使用)
    val imagePaths: String = ""
)

/**
 * 数据项类型枚举
 */
enum class ItemType {
    PASSWORD,    // 密码 (Wear版本不使用)
    TOTP,        // 验证器 (Wear版本主要类型)
    BANK_CARD,   // 银行卡 (Wear版本不使用)
    DOCUMENT,    // 证件 (Wear版本不使用)
    NOTE         // 笔记 (Wear版本不使用)
}

File: Monica for Windows/Monica.Windows/Models/SecureItem.cs
using System;
using System.ComponentModel.DataAnnotations;
using System.ComponentModel.DataAnnotations.Schema;
using CommunityToolkit.Mvvm.ComponentModel;

namespace Monica.Windows.Models
{
    public enum ItemType
    {
        Password = 0, // 兼容
        Totp = 1,
        BankCard = 2,
        Document = 3,
        Note = 4
    }

    [Table("SecureItems")]
    public partial class SecureItem : ObservableObject
    {
        [Key]
        [DatabaseGenerated(DatabaseGeneratedOption.Identity)]
        public long Id { get; set; }

        public ItemType ItemType { get; set; }

        [Required]
        public string Title { get; set; } = string.Empty;

        public string Notes { get; set; } = string.Empty;

        // Encrypted JSON data containing type-specific fields
        public string ItemData { get; set; } = string.Empty;

        public DateTime CreatedAt { get; set; } = DateTime.Now;
        public DateTime UpdatedAt { get; set; } = DateTime.Now;

        [ObservableProperty]
        private bool _isFavorite;

        public int SortOrder { get; set; }

        // TOTP Display Properties (Not Mapped)
        [ObservableProperty]
        [property: NotMapped]
        private string _totpCode = "------";

        [ObservableProperty]
        [property: NotMapped]
        private string _totpTimeRemaining = "";

        [ObservableProperty]
        [property: NotMapped]
        private double _totpProgress = 0;

        [NotMapped]
        public object? CachedTotpData { get; set; }
    }
}
